### PR TITLE
Enable ESLint `no-shadow` and add disable comments

### PR DESCRIPTION
**Pull request by ybiquitous**
This change takes the following steps:

1. Add `no-shadow: error` to our ESLint configuration in `package.json`.
2. Add disable and TODO comments via [`suppress-eslint-errors`](https://github.com/Faithlife/suppress-eslint-errors).

```console
$ npx suppress-eslint-errors lib/**/*.js --message="TODO: Issue #4985"
```

> Which issue, if any, is this issue related to?

This PR is the first step of #4985.

> Is there anything in the PR that needs further explanation?

It is a hard task to fix 100 issues at once, so I think it better that we will fix the issues via multiple PRs.

### CODE REVIEW

**Review by malsf21 — APPROVED** [review]
I like this idea, and the implementation here looks good to me! I'm more than happy to try to contribute some of the issue changes (though there's a backlog of stylelint-related stuff I want to do!). Thanks for the great idea @ybiquitous 

LGTM 👍

File: package.json [hunk 1]
```diff
@@ -61,6 +61,9 @@
     "extends": [
       "stylelint"
     ],
+    "rules": {
+      "no-shadow": "error"
```
Comment (hudochenkov) [inline package.json:65]: I was thinking about adding it to the shared config right away.

Comment (ybiquitous) [reply package.json:65]: OK, I'll open a new PR on <https://github.com/stylelint/eslint-config-stylelint>. 👍 
And, when a new version of `eslint-config-stylelint` will be released, I'll remove this config in `package.json`.

No problems?

Comment (hudochenkov) [reply package.json:65]: Sure!

**Review by hudochenkov — APPROVED** [review]

**Comment by ybiquitous** [comment]
I've opened PR for our shared config: https://github.com/stylelint/eslint-config-stylelint/pull/99

**Comment by ybiquitous** [comment]
I've bumped `eslint-config-stylelint` from 12.1.0 to 13.0.0 and removed needless `rules` in `package.json` (d1bb2ff2):
https://github.com/stylelint/eslint-config-stylelint/releases/tag/13.0.0

**Review by hudochenkov — APPROVED** [review]

**Review by jeddy3 — APPROVED** [review]

**Pull request merged**